tests/test_market.py

```diff
@@ -0,0 +1,431 @@
+        }
+        simulator = test_utils.setup_simulation_entities(
+            config_file=config_file, override_dict=override_dict, agent_policies=agent_policies
+        )
+        return simulator
+
+    def run_market_test(self, agent_policies, expected_wallet, delete_logs=True, full_sim=False):
+        """Test market trades result in the expected wallet balances"""
+        simulator = self.set_up_test(agent_policies=agent_policies)
+        if full_sim:
+            simulator.run_simulation()  # run the simulation
+        else:
+            simulator.collect_and_execute_trades()  # execute one time-step only
+        output_utils.close_logging(delete_logs=delete_logs)
+        # simulation is over, now we inspect the output of the agents' wallets
+        number_of_init_agents = 0  # count number of init agents so we can skip over them
+        for all_agent_index, agent in simulator.agents.items():  # loop over all agents
+            if agent.name == "init_lp":
+                number_of_init_agents += 1
+            else:  # only for custom agents, loop across them and check their parameters
+                custom_agent_index = all_agent_index - number_of_init_agents  # identify which custom agent we are on
+                expected_wallet_dict = expected_wallet[custom_agent_index]
+                for account, expected_balance in expected_wallet_dict.items():  # for each custom parameter to check
+                    agent_balance_in_pts = getattr(agent.wallet, account)
+                    if isinstance(agent_balance_in_pts, dict):
+                        list_of_account_values = list(agent_balance_in_pts.values())
+                        agent_balance_in_pts = sum(x.balance for x in list_of_account_values)
+                    agent_balance_in_base = agent_balance_in_pts * simulator.market.spot_price
+
+                    ##### GENERAL LOGIC TESTING #####
+                    # value of Long position in base has to be less than the # of PTs
+                    # it could only get close if price were equal to 1, and even then there would be slippage
+                    assert agent_balance_in_base < agent.amount_to_trade, (
+                        f"agent #{custom_agent_index}'s {account} is worth more than the amount they traded",
+                    )
+
+                    ##### SPECIFIC MANUAL TESTING #####
+                    # this holds only if we know the exact value
+                    np.testing.assert_almost_equal(
+                        agent_balance_in_pts,
+                        expected_balance,  # passed in as pt's
+                        err_msg=(
+                            f"agent #{custom_agent_index}'s {account} does not equal {expected_balance}"
+                            f", instead we have (base,pt)=({agent_balance_in_base},{agent_balance_in_pts})"
+                        ),
+                    )
+
+    # I'd make this a static method but you can't call assertEqual on a static method
```

Comment:
I don't think the comment about it being a static method is useful in the repo long-term. I'd say take it out if it's all the same to you. The function name should be more descriptive / useful, like `assert_equal_and_log` or something. Type hints should be on all arguments. We (apparently) have much lower standards for docstrings and typehints in tests -- they're not really used anywhere. So I won't ask that you add typehints in this whole file. But in a given function if you're going to do it for one you should just do it for all of them. For me, and my ocd. Thanks!